TSPL.docc/LanguageGuide/Concurrency.md

```diff
@@ -564,33 +637,192 @@ the explicit parent-child relationships between tasks
 let Swift handle some behaviors like propagating cancellation for you,
 and lets Swift detect some errors at compile time.
 
+[`TaskGroup`]: https://developer.apple.com/documentation/swift/taskgroup
+
+<!-- TR: What's a good example of this kind of error? -->
+
+Here's another version of the code to download photos
+that handles any number of photos:
+
 ```swift
 await withTaskGroup(of: Data.self) { taskGroup in
+    let photoNames = await listPhotos(inGallery: "Summer Vacation")
+    for name in photoNames {
+        taskGroup.addTask {
+            let photo = downloadPhoto(named: name)
+            show(photo)
+        }
+    }
+}
+```
```

Comment:
The more usual pattern is:

```swift
    for name in photoNames {
         taskGroup.addTask {
             downloadPhoto(named: name)
         }
     }

for await photo in group { 
  show(photo) 
}
```

so we're showing them one by one, but downloading them in parallel.

It is a bit weird to show a discarding usage like this FIRST -- let's introduce it after we talked about collecting tasks. (This use is unusual, the usual pattern is for `for await`)